### PR TITLE
feat(developer): add api-extractor and api-documenter, config, and build integration 📚

### PR DESCRIPTION
**Pull request by mcdurdin**
Relates to #10207.

Adds @microsoft/api-extractor and @microsoft/api-documenter NPM packages for validating and generating API documentation from Typescript modules.

Adds basic configuration for each module that we will be publishing at this time.

api-extractor's output will be emitted to /developer/docs/api/etc/, for future release comparisons. A baseline output will be committed separately.

The build outputs will be in /developer/build/docs and the intermediate files will be in /developer/build/api.

Note that at this time, we will not enable the checks and validations. That is a task for alpha rather than beta. The primary intent at this time is to select a solid, well-supported toolchain for generating the API documentation for kmc.

Adds the `api` action to build.sh for each affected module. The `api` action for each module will generate the api metadata files, and then the markdown files are generated by /developer/build.sh api.

Note that developer-utils, kmc, and kmcmplib have no api documentation at this time (and kmcmplib is not Typescript so would need separate tooling anyway).

The output of api-documenter is simple and not customizable, but other tooling can be swapped in fairly easily in the future if we prefer. Here's what we get for now:

image

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/build.sh [hunk 1]
```diff
@@ -45,4 +46,11 @@ builder_describe_outputs \
 
 builder_run_action configure cp "$KEYMAN_ROOT/common/resources/fonts/keymanweb-osk.ttf" "$KEYMAN_ROOT/developer/src/tike/xml/layoutbuilder/"
 
-builder_run_child_actions clean configure build test
+builder_run_child_actions clean configure build test api
+
+function build_docs() {
```
Comment (darcywong00) [inline developer/build.sh:51]: Naming question: does this imply a builder action `build:docs` vs the api action?

Comment (mcdurdin) [reply developer/build.sh:51]: Yeah, I should call it build_api...

File: developer/build.sh [hunk 2]
```diff
@@ -45,4 +46,11 @@ builder_describe_outputs \
...
+  api-documenter markdown -i ./build/api -o ./build/docs
+  # TODO: Copy to help.keyman.com and open PR
+}
+
+builder_run_action api build_docs
```
Comment (mcdurdin) [inline developer/build.sh:56]:
```suggestion
function build_api() {
  api-documenter markdown -i ./build/api -o ./build/docs
  # TODO: Copy to help.keyman.com and open PR
}

builder_run_action api build_api
```

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by jahorton** [comment]
Must... avoid... temptation... to try it out immediately...

**Comment by mcdurdin** [comment]
https://help.keyman.com/developer/17.0/reference/api/ you can see the output here (note that this is mostly empty because the APIs are still in the process of being documented!)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.277-beta](https://keyman.com/downloads/releases/beta/17.0.277)

**Pull request merged**